CNV-46346: Adding docs for controller revisions (#99504)
openshift/openshift-docs: modules/virt-about-instance-types.adoc

* A namespaced object: `VirtualMachineInstancetype`
* A cluster-wide object: `VirtualMachineClusterInstancetype`

> Both instance type objects use the same spec.

Reviewer: inplace of spec should we mention VirtualMachineInstancetypeSpec?

Author: It doesn't really fit with our doc guidelines, but since it's not really in scope here I'll just change it back for now.

`launchSecurity`:: Configure Secure Encrypted Virtualization (SEV).
`nodeSelector`:: Specify node selectors to control the nodes where this VM is scheduled.
`schedulerName`:: Define a custom scheduler to use for this VM instead of the default scheduler.

[id="virt-about-instance-types-controller-revisions_{context}"]
== Controller revisions

> When you create a VM by using an instance type, a `ControllerRevision` object retains an immutable, snapshot copy of the instance type object. This snapshot locks in resource-related characteristics defined in the instance type object, such as the required guest CPU and memory. A reference to the `ControllerRevision` object is contained in the VM spec.

Reviewer 2: ```suggestion
When you create a VM by using an instance type, a `ControllerRevision` object retains an immutable, snapshot copy of the instance type object. This snapshot locks in resource-related characteristics defined in the instance type object, such as the required guest CPU and memory. A reference to the `ControllerRevision` object is contained in the VM status.
```

> When you create a VM by using an instance type, a `ControllerRevision` object retains an immutable, snapshot copy of the instance type object. This snapshot locks in resource-related characteristics defined in the instance type object, such as the required guest CPU and memory. A reference to the `ControllerRevision` object is contained in the VM status.

Reviewer 3: ```suggestion
When you create a VM by using an instance type, a `ControllerRevision` object retains an immutable snapshot of the instance type object. This snapshot locks in resource-related characteristics defined in the instance type object, such as the required guest CPU and memory. The VM status also contains a reference to the `ControllerRevision` object.
```

> When you create a VM by using an instance type, a `ControllerRevision` custom resource (CR) retains an immutable, snapshot copy of the instance type spec. This snapshot locks in resource-related characteristics defined in the instance type spec, such as the required guest CPU and memory. A reference to the `ControllerRevision` CR is contained in the VM spec.

Reviewer 2: A `ControllerRevision` isn't a custom resource, if you're seeing it referred to as a `CR` in the upstream docs it's as an acronym for `ControllerRevision`. We also capture the complete object now, just just the spec.

Author: Should it just be "object" instead of custom resource? Or how should we refer to it?

Author: Thank you!

> This snapshot is essential for versioning, ensuring that the VM instance created when starting a VM does not change if the underlying instance type object is updated during the VM's lifetime.

Reviewer 3: ```suggestion
This snapshot is essential for versioning, and ensures that the VM instance created when starting a VM does not change if the underlying instance type object is updated while the VM is running.
```
